make/hotspot/lib/CompileJvm.gmk

```diff
@@ -83,26 +83,25 @@ CFLAGS_VM_VERSION := \
 ################################################################################
 # Disabled warnings
 
-DISABLED_WARNINGS_gcc := parentheses comment unknown-pragmas address \
-    delete-non-virtual-dtor char-subscripts array-bounds int-in-bool-context \
-    ignored-qualifiers  missing-field-initializers implicit-fallthrough \
-    empty-body strict-overflow sequence-point maybe-uninitialized \
-    misleading-indentation cast-function-type shift-negative-value
-
-ifeq ($(call check-jvm-feature, zero), true)
-  DISABLED_WARNINGS_gcc += return-type switch clobbered
-endif
+DISABLED_WARNINGS_gcc := ignored-qualifiers comment int-in-bool-context \
+    array-bounds implicit-fallthrough parentheses missing-field-initializers \
+    delete-non-virtual-dtor unknown-pragmas maybe-uninitialized
+
+ifeq ($(call isTargetCpu, s390x), true)))
+  DISABLED_WARNINGS_gcc += shift-negative-value
+fi
 
-DISABLED_WARNINGS_clang := tautological-compare \
-    undefined-var-template sometimes-uninitialized unknown-pragmas \
-    delete-non-virtual-dtor missing-braces char-subscripts \
-    ignored-qualifiers missing-field-initializers mismatched-tags \
-    shift-negative-value misleading-indentation
+DISABLED_WARNINGS_clang := ignored-qualifiers sometimes-uninitialized \
+    missing-braces delete-non-abstract-non-virtual-dtor unknown-pragmas
```

Comment:
Per JDK-9240259, -Wshift-negative-value should remain globally disabled, for
all of gcc and clang (and xlc, where it hasn't been removed).

Per discussion for JDK-8211073, -Wempty-body should remain globally disabled
for gcc.

As for the other gcc and clang warnings being removed from the globally
disabled lists, that looks okay to me. I'm particularly happy to see
-Wstrict-overflow removed from that list, and that we don't seem to have any
of those any more.

I've not looked at the changes to warnings for Windows.

Reply:
You mentioned shift-negative-value earlier in this PR. I apologize I forgot about it.

I can't find any discussion about JDK-8211073 wrt to `-Wempty-body` (are you sure this is the correct bug ID?), but I can put that as well to the globally disabled set.

As for xlc, I have no access to that compiler and I have therefore not done much changes here. But I think xlc compiles with "warnings as errors" off by default so most likely they don't care.

Reply:
Thanks. I tried searching but my mail archives did not include all of that discussion.

I have now globally disabled empty-body and shift-negative-value for all of hotspot.